Erpett/changelog update1.17 (#17523)
microsoft/vscode-mssql: CHANGELOG.md

# Change Log

## Version 1.17.0
* Release date: January 25, 2023
* Release status: GA

### What's new in 1.17.0

> * BREAKING CHANGE - Encryption is now Enabled by Default - https://github.com/microsoft/vscode-mssql/pull/17484

Reviewer: it may be good to clarify this referring to "Connection encryption" vs. some other encryption, though perhaps that's obvious.

	- Moving to Microsoft.Data.SqlClient 5.0.1 dependency with STS Update

> 	- Connections should now be more sercure but some saved connections may be broken

Reviewer 2: ```suggestion
	- Connections should now be more secure but some saved connections may be broken
```

Reviewer 3: The word "should" makes me itchy. "By default, saved connection profiles will connect with encryption and only accept trusted server certificates. This is a breaking change for some connections and some saved connection profiles may require updates to connect."

Author: Took Drew's updated phrasing

	- See https://aka.ms/vscodemssql-connection for more information. 

> * Introduced HostNameInCertificate Connection Property.

Reviewer 4: @cheenamalhotra Do we have a link to docs to explain this further? Even just the SqlClient docs would be helpful

Author: the link is to a blog post that is going to go live tomorrow

Reviewer 3: @erpett Charles is asking about line 12, not line 11

Author: Oh! I thought it was just because the link isnt live yet. Is there something you all would like to see me add to that line? Another possibility it to make it another sub-bullet with the group above since it was part of that same big PR

Reviewer 3: * Introduced [HostNameInCertificate](https://learn.microsoft.com/dotnet/api/microsoft.data.sqlclient.sqlconnectionstringbuilder.hostnameincertificate#microsoft-data-sqlclient-sqlconnectionstringbuilder-hostnameincertificate) connection property.

(adds a link to the API docs for the property added)

Reviewer 5: ```suggestion
* Introduced 'HostNameInCertificate' Connection Property.
    - This property enables users to set the host name to use when validating the server certificate for the connection.
    - When not specified, the server name from the Data Source is used for certificate validation.
```
You can optionally link, https://learn.microsoft.com/dotnet/api/microsoft.data.sqlclient.sqlconnectionstringbuilder.hostnameincertificate, but it doesn't have much to offer, than what I added above.

* Exposed getServerInfo API to allow Target Platform to be set automatically when creating SQL Projects from database 
	- https://github.com/microsoft/azuredatastudio/issues/20363 
	- https://github.com/microsoft/azuredatastudio/issues/20576
* Accessibility Fixes
	- Fixed Screen Reader not reading status of "loading query" - https://github.com/microsoft/vscode-mssql/issues/17451
	- Fixed Screen Reader to announce messages table in a clearer manner - https://github.com/microsoft/vscode-mssql/issues/17450
	- Fixed issue where Results Table was not minimizable with keyboard - https://github.com/microsoft/vscode-mssql/issues/17452

> ### SQL Grammer Fixes & Updates

Reviewer 2: ```suggestion
### SQL Grammar Fixes & Updates
```

Author: Thanks Chris!

* Fix function syntax highlighting including non-function keywords https://github.com/microsoft/vscode-mssql/pull/17462
* Match "other" keywords last - https://github.com/microsoft/vscode-mssql/pull/17464
* Syntax Update (Mostly from SQL 2022) - https://github.com/microsoft/vscode-mssql/pull/17465
* Add SELECT ALL and FULL/NATURAL JOIN syntax to grammar - https://github.com/microsoft/vscode-mssql/pull/17508

> 	- Special thank you to https://github.com/hanohrs for their suggestion which led to https://github.com/microsoft/vscode-mssql/pull/17508!

Reviewer 4: ```suggestion
	- Special thank you to https://github.com/hanohrs for their help with this!
```
